### ISSUE TITLE
The cursor gets stuck on the right side of the screen

### ISSUE THREAD

**Issue by martincapello** [aseprite/aseprite#4799]
When the selection tool is active and the cursor is moved to the right side of the screen, the cursor freezes. This issue is not intermittent; it occurs 100% of the time.
・These three issues were not present when initially reported, but they started occurring suddenly.
・For example, when connecting a secondary display and performing the same actions on that screen, there are no issues, and the movement works fine.
・I asked a friend to test this issue. On their Windows PC, the problem did not occur. However, on their MacBook Pro (2020, 8GB RAM), the exact same issue occurred.(The OS is both of macOS Sequoia ver.15.1.)

https://github.com/user-attachments/assets/30f57f26-4efa-43af-98e1-5f8bcdec61fe

### Aseprite and System version

* Aseprite version: v1.3.9.2-arm64, Steam
* System: 16-inch MacBook Pro (2021) Specs: 64GB RAM, 8TB storage, macOS Sequoia version 15.1

**Comment by martincapello** [comment]
According to the user that reported this issue, it has been resolved since a few days ago.
Reference: https://github.com/aseprite/aseprite/issues/4787#issuecomment-2563237787

**Comment by martincapello** [comment]
Okay, I'm reopening this, if you have any clue on how to reproduce this, please tell us, because I'm not able to do so. I mean, in the issue description says that when selection tool is active and the cursor moves to the right side of the screen it freezes, tried this and the mouse cursor moves just fine. Now in your new comment the selection tool is not affected...so I'm guessing that we are missing some step to be able to reproduce this issue.

**Comment by mae-1031** [comment]
Thanks @martincapello !
Now this issue occurs regardless of which tool is selected. 
(Previously, it only occurred when the selection tool was active.)

Here are the steps to reproduce the issue:
1. Launch Aseprite.
2. Create a new file.
3. Move the cursor on the canvas.
That’s it. All external devices have been disconnected.

By the way, when I connect an external monitor and use Aseprite on it, the issue does not occur. This means the problem only happens on the MacBook screen. This part is consistent with the previous report.

**Comment by martincapello** [comment]
No problem! 
I did all those steps on my MacBook Pro, 14-inch, 2021, M1 Pro, 16 GB RAM, macOS Sonoma 14.5 and all works fine. I don't have any external device connected as well.
So, I'm not sure if it is something specific to your environment or what...I'm wondering if updating to Sequoia would make any difference...I will test this next week, now I'm working on another issue.

**Comment by martincapello** [comment]
@twoco I've never had this issue...but I didn't update to Sequoia yet...I'm still using Sonoma 14.5 (23F79). Will try to update to Sequoia today to see if anything changes.

**Comment by mae-1031** [comment]
@twoco @martincapello 
The issue continues to occur in my environment as well.
On my MacBook Pro screen, it’s practically impossible to use Aseprite comfortably.
(What bothers me the most is the behavior where the cursor gets stuck on a tab and ends up closing the file!)
However, when I connect an external display and use Aseprite there, the issues are resolved.
I appreciate your continued efforts in investigating this problem.

**Comment by martincapello** [comment]
I've updated to Sequoia 15.4 and I'm still not able to reproduce this...

**Comment by martincapello** [comment]
@twoco Oh, I didn't notice that in your previous video, sorry about that. However, it doesn't occur in my environment...I will pay more attention though. You said that this only happens when connected to a 4K monitor, right? I don't have one to test this...I'll see if I can get one. I've already tested this in a Full HD display connected via HDMI and it works as expected. 
Could you try changing your 4K display to 1920x1080 resolution? Just to confirm if this wobbling issue happens or not on that resolution.

**Comment by martincapello** [comment]
Wait! I was able to reproduce the wobbling shit on my full hd display...now we are talking...I can investigate now

**Comment by martincapello** [comment]
Well, I was able to reproduce it in my macbook display as well...I don't think this issue is related to the mouse stuck issue...but I'll try to fix this one and then you can test if it fixes the mouse stuck

**Comment by mae-1031** [comment]
@martincapello 
In my case, the cursor consistently gets stuck whenever the "Select Tool" is active and I move the cursor to the right side of the canvas. Please continue investigating this issue!

**Comment by mae-1031** [comment]
@martincapello 
I’ve just discovered that the issue disappears if I set Crosshair Type to “Simple Crosshair” and Brush Preview to “None.”
Conversely, with the default settings—“Crosshair on Sprite” and “Full Preview”—the cursor still gets stuck.

**Comment by martincapello** [comment]
Nice finding! That gives us a clue about where to look at. Thank you! 

@twoco Could you try those settings to see if you experience the same results?

**Comment by mae-1031** [comment]
That passage describes exactly the problem I’m seeing.

In my tests, the issue occurs only when I operate Aseprite on the MacBook Pro’s built-in display, and the reproducibility is 100 %.
It is not something that appears over time—the cursor stack happens from the moment Aseprite starts. It occurs when the cursor reaches the right side of the screen, roughly along the line where the F10 and F11 keys are.

Incidentally, if I drag the affected window onto an external monitor without restarting it and perform the same tests, the problem does not occur. This suggests the issue is not caused by cache or similar factors, but rather by the interaction between Aseprite and the MacBook’s internal monitor.

**Comment by mae-1031** [comment]
It seems the reproducibility and the conditions under which the issue appears differ between your setup and mine. Even so, it’s important that we’ve confirmed the cursor-stuck problem definitely exists. Thanks for your response @twoco

**Comment by martincapello** [comment]
Yes, we know that using the simple crosshair and deactivating the brush preview are not more that a kind of workaround...we really want to fix this issue, the thing is that we were not able to reproduce it yet in our environment. So it is really hard for us to find a fix for this in this manner...
I don't think that the renderer is causing the issue...because I should have the same issue, everyone using macOS should have it...could you share how your display is configured? I mean, which resolution, preset, refresh rate for the display where the mouse gets stuck.

@mae-1031 Could you share yours as well?

**Comment by martincapello** [comment]
Which tool have you used?

**Comment by martincapello** [comment]
That's fine, it shouldn't be related to the HDMI. But it is good to know that you are using a tool and having some other issues with the display that might not been related to the issue, but we need as much environment information as possible.

**Comment by martincapello** [comment]
@mae-1031 just by chance, are you using (or have installed) the BetterDisplay app?

@twoco is there any way to disable (unistall) the BetterDisplay app? Because I've been looking into it and it provides several functionalities which seems to have the potential of generating some issues.

**Comment by martincapello** [comment]
> I don't want to reinstall macOS now. I don't have any problems with other apps. Only with Aseprite. 

No worries, I thought it was something easy to turn off. 

> Even if the display settings had something to do with it, why only in Aseprite? 

It should be Aseprite related...I was just suggesting something to try with that tool. The only thing that makes me doubt is that the issue disappears sometimes (in your case, because @mae-1031 case is similar but 100% reproduction rate and the cursor seems to get stuck always in the same place)...

> Is this bug on Macs only? I found a report in forum with similar problems on Windows (4 years ago), but the poster fixed it by driver update. Maybe not really related issue.

As far as I know (and I might be wrong), we have only these two reports, both under macOS. There are some similar reports for Windows as you found, but not exactly the same, mostly performance related or because of some other issue.

Could you please send us an email to support@aseprite.org? And let me know once you did. Thank you.

**Comment by Gasparoken** [comment]
Just in case: It was found in #4681 that another app was interfering with Aseprite (NotchNook.app).
In the current case, it could be due to another similar app. Just in case, it would be good practice to kill apps one by one running in the background and check if the error persists (this would only be practical if the bug is 100% reproducible).
It was also noted that the fix https://github.com/aseprite/laf/pull/131 resolved issue #4681; we hope it works in this case as well.

**Comment by mae-1031** [comment]
Hey, @martincapello 
Sorry I couldn't respond earlier. Due to various circumstances, I was only able to check now.
I updated to v1.3.14.2 a few days ago, and these problems have completely stopped occurring. The stack issues on my MacBook screen are no longer happening. It's back to the excellent user experience it used to be.
Thank you for addressing this so sincerely!
I wanted to send this message to let you know.(I'm grateful to everyone who worked on the improvements.)

**Comment by martincapello** [comment]
Thank you for let us know! Really appreciate it!
